### PR TITLE
Whisper model support in Lite

### PR DESCRIPTION
**Pull request by oyilmaz-nvidia**
# What does this PR do ?

Adds AutoModelSeq2Seq HF support

### CODE REVIEW

File: nemo/collections/llm/gpt/data/hf_dataset.py [hunk 1]
```diff
@@ -61,7 +61,10 @@ def pad_within_micro(batch, pad_token_id):
             max_len = max(map(len, batch))
             return [item + [pad_token_id] * (max_len - len(item)) for item in batch]
 
-        keys = list(filter(lambda x: x in batch[0], ['tokens', 'labels', 'position_ids', 'loss_mask']))
+        print(batch[0])
+        keys = list(
+            filter(lambda x: x in batch[0], ['input_features', 'tokens', 'labels', 'position_ids', 'loss_mask'])
```
Comment (akoumpa) [inline nemo/collections/llm/gpt/data/hf_dataset.py]: any way to avoid this? this change will break LLM peft

Comment (oyilmaz-nvidia) [reply nemo/collections/llm/gpt/data/hf_dataset.py]: Yeah, that was mistakenly pushed. Reverted already.

File: nemo/collections/llm/gpt/data/hf_dataset.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
 from nemo.lightning.pytorch.plugins import MegatronDataSampler
 
 
-class HfDatasetDataModule(pl.LightningDataModule):
+class HFDatasetDataModule(pl.LightningDataModule):
```
Comment (pzelasko) [inline nemo/collections/llm/gpt/data/hf_dataset.py:21]: FYI this rename breaks imports

**Comment by pzelasko** [comment]
I made some changes to enable the training to converge (commit https://github.com/NVIDIA/NeMo/pull/11464/commits/c605d12aa89f8674e6a11141a0e7fdd4c9e80c5c):
* Preserve HF tokenizer's Whisper special tokens (prompt)
* Change the precision to bf16 (fp16 results in nan after the first weight update)
* Fix decoder_input_ids and labels for training step

**Review by titu1994 — APPROVED** [review]
Looks good !

**Pull request merged**